woocommerce/woocommerce-android
# [Blaze] Logic for the MyStore's View when there is no campaign

Closes: #9959 

### Description
This PR adds the logic for the MyStore's Blaze View in the no-campaign state, it adds the following:
1. Logic for fetching campaigns on View launch.
2. Logic for observing the most recent Blaze campaign.
3. When there is no campaign, it handles getting the most recent product and displaying it.

### Testing instructions
1. Please start with a Store with no products.
2. Launch the app, and confirm the Blaze view is hidden.
3. Create a single product and publish it.
4. Go back to the MyStore fragment, and confirm the Blaze View is shown.
5. Click on either the product or the Create Campaign button, and confirm it loads the WebView for Blaze campaign creation for the **specified product**.
6. Dismiss the flow, and go back to the app.
7. Add more products.
8. Go back to the MyStore fragment.
9. Click on Create Campaign, confirm it loads the WebView for Blaze campaign creation for the **site**
10. Go back and click on the product view, and confirm it opens the WebView for Blaze campaign creation for the specified product.
